src/ec/curve25519/ops.rs

```diff
@@ -121,6 +121,79 @@ fn encode_point(x: Elem, y: Elem, z: Elem) -> EncodedPoint {
     bytes
 }
 
+// If the target architecture is non-Windows x86-64 or NEON-capable ARM, it is
+// faster to do scalar base point multiplication using the generic scalar
+// multiplication implementation.
+#[cfg(all(target_arch = "x86_64", not(windows)))]
+pub fn scalar_mult_with_base_point(scalar: &Scalar) -> EncodedPoint {
```

Comment:
> This is an EncodedPoint in the X25519 sense, not the Ed25519 sense, right?

Exactly.

> It would be nice to find a way to make this clearer.

My plan has been, after the existing changes were approved, to add more commits which add a statically-checked type-level distinction between the different kinds of encoded points, and between cases where we pass true `Scalars` (masked) and byte arrays that _will become_ scalars (still need to be masked). My understanding is that there was a little bit of uncertainty still about how much of a win some of those tactics were in the RSA/Montgomery arithmetic work, so I held off to avoid dumping a ton of commits on you for review.

I'll make a run at that _after_ I loop back and address your other comments. At a minimum, we can improve the use of aliases.